src/utils/date-time.js

```diff
@@ -45,81 +39,77 @@ export const isProperTimezone = (timeZone: string): boolean => {
   return true
 }
 
+export const getDateWithOffset = (date, offset) => moment(date).utcOffset(offset)
+
 export const useDateTime = () => {
   const timezone = useSelector(selectTimezoneSetting)
+  const utcOffset = useSelector(selectUTCOffsetSetting)
+
   const isUsingTimezone = timezone !== "" && timezone !== "default"
 
   const localeDateString = useMemo(() => {
-    const dateOptions = {
+    const dateOptions = long => ({
+      hourCycle: 'h23',
       localeMatcher: "best fit",
       formatMatcher: "best fit",
-      weekday: "short",
-      year: "numeric",
-      month: "short",
-      day: "2-digit",
+      ...(long
+        ? { weekday: "short", year: "numeric", month: "short", day: "2-digit" }
+        : { dateStyle: "short" }),
       timeZone: isUsingTimezone ? timezone : undefined,
-    }
-    const dateFormat = () => new Intl.DateTimeFormat(navigator.language, dateOptions)
+    })
+    const dateFormat = long => new Intl.DateTimeFormat(navigator.language, dateOptions(long))
     return isSupportingDateTimeFormat
-      ? (d: Date | number) => dateFormat().format(d)
+      ? (date, long = true) => dateFormat(long).format(date)
```

Comment:
In general is better to pass flag arguments as object, so we know what is it about on call.
```
localeDateString(endDate, {long: false})
```